### ISSUE TITLE
Create env flag for watch

### ISSUE THREAD

**Issue by BrynCooke** [apollographql/router#766]
Currently the user has to enable hot reloading by explicitly setting a `--watch` arg when running the router.

This is difficult for users to do, whereas we could expose an env variable to enable it so that uses can do this kubernetes config.

All this means is adding the `env` attribute to the clap config for `watch`.

**Comment by BrynCooke** [comment]
There is some discussion around this is detrimental for prod Kubernetes environments.

Updating configuration in Kubernetes is a hot topic (pun intended), and there are lots of different solutions because kubernetes did not always have rollout capability.

That being said I am still in favor of enabling the hot reload by default if we can make it work. Here are the reasons why:

* Built in rollout functionality is not affected adversly by hot reload. The workflow for this is that the user should create a new config map and update their deployment to point to this. Effectively users must treat configmaps as immutable. This allows them to roll back if needed.
* Larger deployments will probably not be using built in kubernetes rollout at all and instead be using a third party product whihc will enforce the immutability of configmaps.
* In the case where users do decide to not treat their config maps as immutable, *not* having hot reload will allow the running state of the application to diverge from the configured state. This will lead to a confusing experience where a pod may show that it has a certain config, but in fact it is running something else because it has not been restarted.
* It's better during dev and small kubernetes deployments where users are prepared to just use hot reload.

The TLDR is that I think hot reload of config is still better for prod than the alternative of letting things get out of sync, and if users are advanced enough to be using rollouts then it doesn't hurt them either.
